ddtrace/contrib/cherrypy/middleware.py

```diff
@@ -0,0 +1,149 @@
+"""
+Datadog trace code for cherrypy.
+"""
+
+# stdlib
+import logging
+
+# project
+from ddtrace import config
+from .. import trace_utils
+from ... import compat
+from ...ext import errors, SpanTypes
+from ...propagation.http import HTTPPropagator
+from ...utils.formats import asbool, get_env
+
+# 3p
+import cherrypy
+from cherrypy.lib.httputil import valid_status
+
+log = logging.getLogger(__name__)
+
+
+# Configure default configuration
+config._add(
+    "cherrypy",
+    dict(
+        distributed_tracing=asbool(get_env("cherrypy", "distributed_tracing", default=True)),
+    ),
+)
+
+SPAN_NAME = "cherrypy.request"
+
+
+class TraceTool(cherrypy.Tool):
+    def __init__(self, app, tracer, service, use_distributed_tracing):
+        self.app = app
+        self._tracer = tracer
+        self.service = service
+        self.use_distributed_tracing = use_distributed_tracing
+
+        # CherryPy uses priority to determine which tools act first on each event. The lower the number, the higher
+        # the priority. See: https://docs.cherrypy.org/en/latest/extend.html#tools-ordering
+        cherrypy.Tool.__init__(self, "on_start_resource", self._on_start_resource, priority=95)
+
+    @property
+    def use_distributed_tracing(self):
+        return config.cherrypy.get("distributed_tracing", True)
+
+    @use_distributed_tracing.setter
+    def use_distributed_tracing(self, use_distributed_tracing):
+        config.cherrypy["distributed_tracing"] = asbool(use_distributed_tracing)
+
+    @property
+    def service(self):
+        return config.cherrypy.get("service", "cherrypy")
+
+    @service.setter
+    def service(self, service):
+        config.cherrypy["service"] = service
+
+    def _setup(self):
+        cherrypy.Tool._setup(self)
+        cherrypy.request.hooks.attach("on_end_request", self._on_end_request, priority=5)
+        cherrypy.request.hooks.attach("after_error_response", self._after_error_response, priority=5)
+
+    def _on_start_resource(self):
+        if self.use_distributed_tracing:
+            propagator = HTTPPropagator()
+            context = propagator.extract(cherrypy.request.headers)
+            # Only need to active the new context if something was propagated
```

Comment:
```suggestion
            # Only need to activate the new context if something was propagated
```

Pretty sure this was copy-paste as I've seen this typo before in the repo 😆

Reply:
:+1:  Copy and paste. Best friend and worst enemy!